theano/gof/graph.py

```diff
@@ -117,6 +117,9 @@ def __init__(self, op, inputs, outputs):
                 if output.owner is None:
                     output.owner = self
                     output.index = i
+                    if not hasattr(output, "tag"):
+                        output.tag = utils.scratchpad()
+                    utils.add_tag_trace(output)
```

Comment:
Why this line is needed? We already do this at many other places. It should be done for all Constant I think. It not, tell me. For shared variable, I found a problem in our logic. Currently, we set it only when we call theano.shared(), but not when we call directly the type specific shared variable constructor like gpuarray_shared_variable as done in many tests.

Is that the case you found? If so, I don't think this is the good fix. This will add the stack-trace as of when we use the shared variable, not when we create it. I think the fix should be to call add_tag_trace in SharedVariable.__init__() and remove it from theano/compile/sharedvalue.py:shared()